### PR TITLE
Removed deprecation warnings from `train_utils.py`

### PR DESCRIPTION
**Pull request by jupyterjazz**
**Proposed changes**:
- Removed deprecation warnings in `rasa/utils/train_utils.py` for the 3.0 release.  #9399 

**Status (please check what you already did)**:
- ~[ ] added some tests for the functionality~
- [x] the deprecation warnings are resolved
- [x] updated the documentation
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

**Review by samsucik — CHANGES_REQUESTED** [review]
@jupyterjazz we're getting there, just a few things that need attention:
- We need to remove the `DOCS_URL_MIGRATION_GUIDE_WEIGHT_SPARSITY` constant from everywhere
- There's a passage including "Used inside softmax loss only." in the comments in all of our ML models, this needs to be changed to talk about the cross-entropy loss instead
- Finally, I think that this PR shouldn't make any changes to the "underscored" modules (e.g. `_diet_classifier.py`). I don't know much about what the purpose of those modules is, but at the very top of the files it says `Please don't do any changes in this module`.

**Comment by samsucik** [comment]
@jupyterjazz as for the 3 failing checks, I admit that they're a big mystery to me... Let me know if you want to work together on investigating it. But maybe Tomasz can help you as well?

**Comment by jupyterjazz** [comment]
@samsucik, thanks for the suggestions! I kept files with underscores unchanged, but apparently, they're used in tests, and as I've removed some of the functions from the `train_utils.py` these tests fail.

attributeerror

Should I reach out to people responsible for the underscored files and ask them for help/clarification?

**Comment by samsucik** [comment]
@jupyterjazz good to know (regarding the underscored modules). Yeah, I think your best bet is to reach out to Enable to find out how exactly those modules are supposed to be treated these days. Let me know what you've found out, please 🙂

**Comment by jupyterjazz** [comment]
@samsucik, here's my understanding of what's happening:
As the new graph architecture is being currently integrated, the original files (e.g. `diet_classifier.py`) contain classes adapted to the new architecture (e.g. `DIETClassifierGraphComponent`) and the old classes (`DIETClassifier`) are in the underscored files. As the integration is not yet completed, the tests are using the old classes and that's why they were failing. The integration is expected to finish in the next week or so.  
I talked to @wochinge and he approved changing the underscored files.

**Review by samsucik — APPROVED** [review]
Looks good to me, thanks @jupyterjazz!

**Review by wochinge — APPROVED** [review]
Thanks for reaching out and communicating proactively! 💯
